lib/jt400.ts

```diff
@@ -196,14 +227,30 @@ function createInstance(connection, insertListFun, inMemory) {
           return row.map(convertDateValues)
         })
       )
-      return Q.nfcall(thisConn.batchUpdate, sql, jsonParams).then(res =>
-        Array.from(res)
-      )
+      return Q.nfcall(thisConn.batchUpdate, sql, jsonParams)
+        .then(res => Array.from(res))
+        .catch(error => {
+          throw newProgrammerOops(
+            error.stack.split('\n')[1],
+            {
+              jt400FunctionCall: 'batchUpdate',
+              sql,
+              params: JSON.parse(jsonParams)
```

Comment:
Nitpicking. But I would rather create a const params that is not stringified and skip JSON.parse here :)

Reply:
Totally agree.